### PR TITLE
MULE-12609: Every exported Mule package must contain api

### PR DESCRIPTION
**Pull request by pablokraan**

### CODE REVIEW

File: core/src/main/resources/META-INF/mule-module.properties [hunk 1]
```diff
@@ -47,6 +51,7 @@ artifact.export.classPackages=org.mule.runtime.core,\
                               org.mule.runtime.core.api.security,\
                               org.mule.runtime.core.api.security.tls,\
                               org.mule.runtime.core.api.serialization,\
+                              org.mule.runtime.core.api.session,\
```
Comment (elrodro83) [inline core/src/main/resources/META-INF/mule-module.properties]: Shouldn't all session stuff be privileged api?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java [hunk 2]
```diff
@@ -37,9 +37,9 @@
 import org.mule.runtime.core.api.source.MessageSource;
 import org.mule.runtime.core.api.transaction.TransactionConfig;
 import org.mule.runtime.core.exception.ErrorTypeLocator;
-import org.mule.runtime.core.execution.ExceptionCallback;
-import org.mule.runtime.core.execution.MessageProcessContext;
-import org.mule.runtime.core.execution.MessageProcessingManager;
+import org.mule.runtime.core.internal.execution.ExceptionCallback;
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java]: shouldn't ExtensionMessageSource be api?

File: core/src/main/resources/META-INF/mule-module.properties (hunk 1 above)
Comment (pablokraan) [inline core/src/main/resources/META-INF/mule-module.properties]: It must be api, functional tests depend on it

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java (hunk 2 above)
Comment (pablokraan) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java]: Is not required at API level, is only used by extension support modules

**Pull request merged**